File: ePrijevozSarajevo/ePrijevozSarajevo.Model/IssuedTicket.cs
namespace ePrijevozSarajevo.Model
{
    public class IssuedTicket
    {
        public int IssuedTicketId { get; set; }
        public User User { get; set; } = null!;
        public int UserId { get; set; }
        public Ticket Ticket { get; set; } = null!;
        public int TicketId { get; set; }
        public DateTime ValidFrom { get; set; }
        public DateTime ValidTo { get; set; }
        public DateTime IssuedDate { get; set; } = DateTime.Now;
        public int? Amount { get; set; }
        public Route? Route { get; set; }
        public int? RouteId { get; set; }

    }
}

File: ePrijevozSarajevo/UI/eprijevoz_desktop/lib/models/issuedTicket.dart
import 'package:json_annotation/json_annotation.dart';

part 'issuedTicket.g.dart';

@JsonSerializable()
class IssuedTicket {
  int? issuedTicketId;
  int? userId;
  int? ticketId;
  DateTime? validFrom;
  DateTime? validTo;
  DateTime? issuedDate;
  int? amount;
  int? routeId;

  IssuedTicket(
      {this.issuedTicketId,
      this.userId,
      this.ticketId,
      this.validFrom,
      this.validTo,
      this.issuedDate,
      this.amount,
      this.routeId});

  factory IssuedTicket.fromJson(Map<String, dynamic> json) =>
      _$IssuedTicketFromJson(json);

  Map<String, dynamic> toJson() => _$IssuedTicketToJson(this);
}
